coreneuron/sim/fadvance_core.cpp

```diff
@@ -108,11 +108,37 @@ void finalize_progress_bar() {
     }
 }
 
-void nrn_fixed_step_group_minimal(int n) {
-    static int step = 0;
+void nrn_fixed_single_steps_minimal(int total_steps, double tstop) {
```

Comment:
Just to be sure : `total_steps` here is number of steps in single mindelay / communication interval and not whole simulation?

Reply:
It's the `total_steps` of the whole simulation. `nrn_fixed_single_steps_minimal` and `nrn_fixed_step_group_minimal` are called by `ncs2nrn_integrate` with the total number of steps, calculated based on the `tstop` given as argument of the CLI of `CoreNeuron`. I will rename some parameters to make it clearer. :+1: